docs/DesignDocs/ExtendedEnclaveInitializationData_V2.md

```diff
@@ -0,0 +1,424 @@
+     *  post enclave initialization. Currently only supported by SGX Enclaves
+     *  with KSS feature enabled or SGX EEID enclaves.
+     */
+    OE_ENCLAVE_SETTING_CONTEXT_PRE_MEASURED_ID = 0x976a8f66,
+    /** Heap, stack, and thread configuration for an dynamic sizing SGX EEID
+     *  enclave instance. Only supported if the base enclave is a dynamic
+     *  sizing EEID enclave.
+     */
+    OE_ENCLAVE_SETTING_CONTEXT_EEID_DYNAMIC_SIZES = 0x976a8f67,
+    /** Additional content to be loaded into the enclave automatically by the
+     *  runtime post enclave initialization. Currently only supported by SGX
+     *  Enclaves with KSS feature enabled or SGX EEID enclaves.
+     */
+    OE_ENCLAVE_SETTING_CONTEXT_EEID_PRE_MEASURED_DATA = 0x976a8f68,
+#endif
+} oe_enclave_setting_type_t;
+/**
+ * Structure to keep EEID related options during enclave creation
+ */
+typedef struct _oe_enclave_pre_measured_id
+{
+    /** The identity of additional content allowed to be loaded
+     *  into the enclave post enclave initialization. The identity is covered
+     *  by TEE generated Enclave Attestation Evidence.
+     */
+    uint8_t config_id[64];
+    uint16_t config_svn;
+}
+oe_enclave_pre_measured_id_t;
+typedef struct _oe_enclave_eeid_dynamic_sizes
+{
+    /** Heap, stack, and thread configuration for an EEID enclave instance. */
+    oe_enclave_size_settings_t size_settings;
+}
+oe_enclave_eeid_dynamic_sizes_t;
+typedef struct _oe_enclave_eeid_pre_measured_data
+    /** Config data to be loaded automatically by the enclave runtime.
+     * If provided, the enclave runtime overrides config_id field with the
+     * SHA256 hash of the config data, and verifies the hash value when the
+     * data is loaded.
+     */
+    size_t data_size;
+    uint8_t data[];
+} oe_enclave_eeid_pre_measured_data_t;
+```
+
+Interaction of settings
+-----------------------
+
+Based on the type of the base enclave image, the behavior of each of the
+combinations of settings is as follows (x meaning provided):
+
+*Non-EEID SGX Enclave*
+
+| DYN | PREID | PREDATA | Behavior
+|-----|-------|---------|-----------------------------------
+|  -  |   -   |   -     | On system where SGX-KSS feature is not available or disabled: N/A; On system with SGX-KSS enabled: loader sets SECS.config_id and SECS.config_svn as 0
+|  -  |   x   |   -     | On system where SGX-KSS feature is not available or disabled: Invalid;  On system with SGX-KSS enabled only: loader copies PREID to SECS.config_id and SECS.config_svn
+
+*Static sizing EEID SGX Enclave*
+
+| DYN | PREID | PREDATA | Behavior
+|-----|-------|---------|-----------------------------------
+|  -  |   -   |   x     | Static sizing EEID enclave only: loader sets eeid_page.config_id as PREDATA hash, and eeid_page.config_svn as 0, enclave runtime loads/verifies PREDATA
+|  -  |   x   |   -     | static sizing EEID enclave only: loader copies PREID to SECS.config_id and SECS.config_svn
```

Comment:
Per SIG-Arch meeting, can you correct the wording? This doesn't write to SECS.config_* fields.